### ISSUE TITLE
Regressions caused by #5315

### ISSUE THREAD

**Issue by kd-11** [RPCS3/rpcs3#5380]
Tracker for any issues caused by https://github.com/RPCS3/rpcs3/pull/5315.
Identified issues so far:
- [x] Some users have significantly reduced performance in some Rachet & Clank titles.
- [ ] Some AMD cards have performance regressions in some applications.
- [x] OpenGL renderer has worse performance in some applications.
- [x] Null renderer is broken.

**Comment by digtal-devil** [comment]
Ninja Gaiden Sigma freezes on the main menu while the music plays in the background. 

[Log](http://www.mediafire.com/file/da3zsspqb1h83zr/RPCS3log.rar/file)

**Comment by Dani1977** [comment]
hi
Demon's souls crash on nexus
 [RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2635701/RPCS3.zip)

**Comment by isshininu** [comment]
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2635861/RPCS3.log.gz)
![1532](https://user-images.githubusercontent.com/6324523/49328252-ab1c9500-f586-11e8-850e-a95ab9a4c96f.png)
![1533](https://user-images.githubusercontent.com/6324523/49328253-ab1c9500-f586-11e8-9e6f-cb5a16e7aefe.png)

Found unusual bug - image in DeS could become black for 1-2 frames, like write color buffers was disabled.

**Comment by altiereslima** [comment]
Reduced performance in Minecraft (R9 390, Vulkan Driver 2.0.55)
#5372 
![rapido](https://user-images.githubusercontent.com/18044879/49329689-e0ff5080-f569-11e8-824c-1cc53f4e41a4.png)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2635971/RPCS3.log.gz)


#5379 **RSX always at 99%**
![lento](https://user-images.githubusercontent.com/18044879/49329714-861a2900-f56a-11e8-85d3-a5e1ff584646.png)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2635977/RPCS3.log.gz)

**Comment by altiereslima** [comment]
Thank you, kd-11

**Comment by kd-11** [comment]
@isshininu Duplicate of https://github.com/RPCS3/rpcs3/issues/4857? Afaik the white flickering was fixed but the black flickering remains. Its not trivial to fix it and it occurs somewhat rarely. Will be fixed separately once the new texture cache changes are merged.

**Comment by kd-11** [comment]
@altiereslima Can you gather a list of affected applications when using AMD+Vulkan? It would be nice to have some more comparison data since 1 game is not enough of a sample size. Also, check how OGL was affected by the changes. I have noted some performance loss on AMD vulkan, but still not enough for the profiler to show any usable differences.

**Comment by isshininu** [comment]
I cleared all cache and didn't see any more black flickering.
By the way, Demon's Souls perfomance on Vulkan in lower than on OpenGL now, especially sound stutters that were introduced in #5315 
It looks like each "RSX: Fault in uninterruptible code!" line breaks background SFXs.

**Comment by Wizzy2018** [comment]
God Of War 3 has now lower fps on cutscenes from 60fps to 25fps on RX470

**Comment by altiereslima** [comment]
> @altiereslima Can you gather a list of affected applications when using AMD+Vulkan? It would be nice to have some more comparison data since 1 game is not enough of a sample size. Also, check how OGL was affected by the changes. I have noted some performance loss on AMD vulkan, but still not enough for the profiler to show any usable differences.

Only Minecraft was affected using AMD+Vulkan.
OpenGL increase 30% in Minecraft.
OpenGL increase 2-4 fps in other games.

**Comment by kd-11** [comment]
@Wizzy2018 Need before/after screenshots so that testers can duplicate the exact scenario.

**Comment by Wizzy2018** [comment]
@kd-11 ok i send you thx for a help :)

**Comment by Dani1977** [comment]
hi, i made a clean installation of RPCS3, but emulator keep crash random, with error 

here an example of crash on training area

![immagine](https://user-images.githubusercontent.com/31347274/49341732-e33cda00-f651-11e8-89c2-d9641eae23f4.png)

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2637082/RPCS3.zip)

**Comment by kd-11** [comment]
That message does not mean crashing. Its just a diagnostic.

**Comment by Dani1977** [comment]
this is my event view log:
Il programma rpcs3.exe versione 0.0.0.0 non interagisce più con Windows ed è stato chiuso. Per vedere se sono disponibili ulteriori informazioni sul problema, verificare la cronologia del problema in Sicurezza e manutenzione nel Pannello di controllo.
 ID processo: 9bc
 Ora di avvio: 01d48a55266ccb72
 Ora di chiusura: 5
 Percorso applicazione: G:\rpcs3\rpcs3.exe
 ID segnalazione: 7ea531f1-2758-4285-a675-3b6791643507
 Nome completo pacchetto che ha generato l'errore: 
 ID applicazione relativo al pacchetto che ha generato l'errore: 
 Tipo interruzione: Unknown

RPCS3 Hang-on for a minute than crash.
Regards

**Comment by Wizzy2018** [comment]
@kd-11 i found what was a problem in my case :D After i disabled a SPU loop detection my fps is back to normal and everything works well

**Comment by Dani1977** [comment]
hi,
I clean, spu, ppu and shader cache than I play Demon's souls around 2 hours without a crash.
I change Preferred SPU Threads from auto to 4, a little bit of drops of frames but no freeze and crash.

Regards

**Comment by cesarnox** [comment]
intel i5 8400 integrated graphics HD 630 

vulkan get stuck at this point 
![captura](https://user-images.githubusercontent.com/21127483/49344124-13c44a00-f639-11e8-81b8-1c0f05fe8cea.PNG)

if I push quickly pass this screen and then stuck and the sound continue for a while 
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2637310/RPCS3.log.gz)

using OpenGL works and faster than before 

![captura2](https://user-images.githubusercontent.com/21127483/49344247-2c812f80-f63a-11e8-8a5a-2037536990fe.PNG)

graphics issues on  are not related to #5315

**Comment by isshininu** [comment]
> Duplicate of #4857? Afaik the white flickering was fixed but the black flickering remains. Its not trivial to fix it and it occurs somewhat rarely. Will be fixed separately once the new texture cache changes are merged.

Yep, this is return of old issue. Only on OpenGL renderer - i didn't see this issue on Vulkan yet.

**Comment by kd-11** [comment]
Oddly, I've always had this DeS issue, it never went away for me ever since I changed my CPU a year ago. So it could just be the different timings manifesting. It warrants investigation on its own.

**Comment by digtal-devil** [comment]
> For example Ninja gayden works for me.

I deleted the shader cache but the game will still hang on the main menu if the Vulkan Renderer is used. I can play NGS if I switch to OpenGL. Oddly enough, the graphics in certain areas have some visual glitches.

**Comment by Dani1977** [comment]
Hi,
i have update rpcs3 to version v0.0.5-7546 (clean install), but unfortunately i have the same game freeze than emulator crash.
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2640779/RPCS3.zip)

but event viewer is difference than before, now indicate ucrtbase.dll version 10.0.17763.1

this is the event (sorry it is in Italian)

Nome dell'applicazione che ha generato l'errore: rpcs3.exe, versione: 0.0.0.0, timestamp: 0x5c056cc5
Nome del modulo che ha generato l'errore: ucrtbase.dll, versione: 10.0.17763.1, timestamp: 0x309241e0
Codice eccezione: 0xc0000409
Offset errore 0x000000000006f08e
ID processo che ha generato l'errore: 0x3748
Ora di avvio dell'applicazione che ha generato l'errore: 0x01d48b36899300cb
Percorso dell'applicazione che ha generato l'errore: G:\rpcs3\rpcs3.exe
Percorso del modulo che ha generato l'errore: C:\WINDOWS\System32\ucrtbase.dll
ID segnalazione: 92280b90-aa32-41f1-8e9c-567bc3c52514
Nome completo pacchetto che ha generato l'errore: 
ID applicazione relativo al pacchetto che ha generato l'errore: 

regards

**Comment by Daist** [comment]
Sacred 2 regressed in PR5315, see below comparison with the previous master:
https://i.imgur.com/MMiCtqy.jpg
PR5391 looks the same as in PR5315.
https://i.imgur.com/BljtISc.jpg

**Comment by HenningJW** [comment]
@Daist can you post logs?

**Comment by Daist** [comment]
@HenningJW Sure
[7513-RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2641121/7513-RPCS3.log.gz)
[7536-RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2641122/7536-RPCS3.log.gz)
[7546-RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2641123/7546-RPCS3.log.gz)

**Comment by NotUnknownbrackets** [comment]
F {rsx::thread} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Out of pool memory (VK_ERROR_OUT_OF_POOL_MEMORY_KHR)
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkgsrender.cpp:1058)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2657101/RPCS3.log)

**Comment by LunarisUmbra** [comment]
Not sure if this is the proper place to post this, but launching Nier Replicant and selecting 'White Color Buffers' to fix white hue overlay on graphics rendering of the game causes audio stutter that was otherwise not there. If there is something needed like a log or video example I'll be glad to supply it.

**Comment by isJuhn** [comment]
R&C ACiT still runs like dogshit after minutes of gameplay on vulkan even after #5391

**Comment by psennermann** [comment]
Maybe #5515 should be added to this issue...

**Comment by kd-11** [comment]
Closing meta-tracker, if any regressions are left (Sacred?) open a ticket for them so they can be investigated quickly.
